File: Journey.Core/ViewModels/Chapter.cs
using Journey.Models.DTO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Journey.Core.ViewModels
{
    public class Chapter : IIndexedEntity
    {
        [JsonPropertyName("pk")]
        public string Pk { get; set; }
        [JsonPropertyName("title")]
        public string Title { get; set; }
        [JsonPropertyName("body")]
        public string Body { get; set; }
        [JsonPropertyName("userId")]
        public string UserId { get; set; }
        [JsonPropertyName("tags")]
        public IEnumerable<string> Tags { get; set; }
        [JsonPropertyName("sequence")]
        public int Sequence { get; set; }

        public string IndexName => "user_chapters";

        public string ItemName => "chapters";
    }
}

File: Journey.UI/journeyApp/src/app/models/chapter.ts
import { Tag } from "./tag";

export class Chapter {
    pk:string;
    id:number;
    sequence:number;
    title:string;
    body:string;
    createdAt:Date;
    tags:Tag[];

    constructor() {
        this.createdAt = this.createdAt || new Date();
        this.tags = new Array();
    }
}
